ludwig/models/predictor.py

```diff
@@ -65,23 +66,33 @@ class Predictor(BasePredictor):
 
     def __init__(
         self,
-        model: BaseModel,
+        dist_model: nn.Module,
         batch_size: int = 128,
         distributed: DistributedStrategy = None,
         report_tqdm_to_ray: bool = False,
+        model: Optional[BaseModel] = None,
         **kwargs,
     ):
+        model = model or dist_model
+        assert isinstance(model, BaseModel)
+
         self._batch_size = batch_size
         self._distributed = distributed if distributed is not None else LocalStrategy()
         self.report_tqdm_to_ray = report_tqdm_to_ray
 
-        # TODO (jeffkinnison): revert to using the requested device for GBMs when device usage is fixed
-        self.device = get_torch_device() if not model.type() == MODEL_GBM else "cpu"
-        self.model = model.to(self.device)
+        device = get_torch_device()
+        if model.type() == MODEL_GBM:
+            # TODO (jeffkinnison): revert to using the requested device for GBMs when device usage is fixed
+            device = "cpu"
+            dist_model = dist_model.to(device)
+
+        self.device = device
+        self.dist_model = dist_model
```

Comment:
Can you leave a comment on the difference between these two models? Maybe as part of an overall class docstring